sushichef.py

```diff
@@ -1,33 +1,264 @@
+        scrape_spreadsheet()
+        for grade in GRADE_DICT:
+            source_id = grade.strip().replace(" ", "_")
+            LOGGER.info("\tCreating a topic node - {}".format(grade))
+            topic = nodes.TopicNode(source_id=source_id, title=grade.capitalize())
+            get_nodes_from_dict(topic, GRADE_DICT[grade], grade)
+            channel.add_child(topic)
+        raise_for_invalid_channel(channel)  # Check for errors in channel construction
+        return channel
+
+def get_nodes_from_dict(parent, dictionary, prefix):
+    """
+    Strucutre nodes from dictionary to connect them
+    """
+    for topic in dictionary:
+        if isinstance(topic, str):
+            topic_source_id = "{}-{}".format(prefix, topic.strip().replace(" ", "_"))
+            topic_node = nodes.TopicNode(source_id=topic_source_id, title=topic.capitalize())
+            LOGGER.info("\tCreating a math topic node - {}".format(topic))
+            get_nodes_from_dict(topic_node, dictionary[topic], topic_source_id)
+            parent.add_child(topic_node)
+        else:
+            for node in topic:
+                if node and not isinstance(node, str):
+                    LOGGER.info("\tAdding a child node - {}".format(node))
+                    parent.add_child(node)
+
 
-class BetterWorldEdChef(LineCook):
+def get_info(information):
     """
-    Sushi chef for uploading the Better World Ed content folders generated by `souschef.py`.
+    Retrieves title of the source
+    For example, from the information below,
+    HYPERLINK("https://drive.google.com/open?id=0B9q-Bz2y-5byRWZwRHptZmk0eU0","Baby And Her Health")
+    Returns `Baby And Her Health`
+    """
+    information = information.lower()
+    if information.startswith("=hyperlink"):
+        splits = information.split(",")
+        return splits[1].split("\"")[1].strip()
+    return information.strip()
+
+def download_video(link):
     """
-    pass  # no custom code needed, just using generic `LineCook`
+    downloads video from the link
+    and returns video node
+    """
+    if "=HYPERLINK" not in link:
+        return link
+    splits = link.split(",")
+    video_link = splits[0].split("=HYPERLINK(\"")[1][:-1]
+    video_id = video_link.split("/")[-1]
+    ydl = youtube_dl.YoutubeDL({
+        'outtmpl': './downloads/%(id)s.%(ext)s',
+        'writeautomaticsub': True
+    })
+
+    with ydl:
+        try:
+            result = ydl.extract_info(video_link)
+        except:
+            LOGGER.info("\tThere was an error while downloding video - {}".format(video_link))
+            return None
+
+    if 'entries' in result:
+        video = result['entries'][0]
+    else:
+        video = result
 
+    video_title = video["title"].capitalize()
+    video_source_id = video_title.strip().replace(" ", "_")
+    video_path = "{}/{}.mp4".format(DOWNLOAD_DIRECTORY, video_id)
+    video_file = files.VideoFile(path=video_path, language=languages.getlang('en').code)
```

Comment:
I did a spot check of one video,
http://35.196.143.130/learn/#/topics/c/64a70c91c5fd5f1299b3ef83f0a7503b ,
and I find it is pretty large 100MB+ with resolution 1920x900

We might want to consider downloading a lower-res version from youtube.

Reply:
Would it be sufficient to just add the compress flag?

Reply:
I ran the chef again with `--compress` flag and it seems that videos become smaller now.